addons/binding/org.openhab.binding.km200/ESH-INF/thing/device.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="km200"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+   <bridge-type id="kmdevice">
```

Comment:
Is a KM device capable of an arbitrary number of `dhwCircuit`, `gateway`, `heatSource` and `heatingCircuit` things? I´m just wondering why the KM device is the bridge which holds possibly an unforeseeable number of sub-things.
As a user I would expect the main thing to be my specific heating system with all its functionality being the channels.
One aspect which supports my thesis is that none of the other thing-types do have a configuration parameter which identifies them specifically. Lets say the bridge kmdevice holds more then one heatSource. How is the binding supposed to differentiate between them?
Your current modelling seems to use the bridge as a grouping instance for the different hardware components inside the heating system. The better approach would be to have the complete heating system as one thing-type, at best one specific device type. So you would have three thing types: km200, km100 and km50 with each of them providing channels (which may be grouped with channel-groups) which express specific functionality of the heating system. Please have a look at the development documentation of the underlying framework (Eclipse SmartHome) on how bridges, things and channels should be modelled: https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html

Reply:
This bridge has a combination of things. Every bridge is a bit different and their combination of things and parameters is different too. Things like dhwCircuit, gateway, heatSource and heatingCircuit are existing always once. But like I wrote, on almost every device with different parameters. The brige is reading the devices capabilities and configurations. All things are created full automaticaly depending on real capabilities of the heating device.